plugins/outputs/opentsdb/opentsdb_http.go

```diff
@@ -165,14 +174,11 @@ func (o *openTSDBHttp) flush() error {
 		_, _ = io.Copy(io.Discard, resp.Body)
 	}
 
-	if resp.StatusCode/100 != 2 {
-		if resp.StatusCode/100 == 4 {
-			log.Printf("E! Received %d status code. Dropping metrics to avoid overflowing buffer.",
-				resp.StatusCode)
-		} else {
-			return fmt.Errorf("Error when sending metrics. Received status %d",
-				resp.StatusCode)
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		if resp.StatusCode < 400 || resp.StatusCode > 499 {
```

Comment:
The previous logic was checking for anything in the 400s right? So shouldn't this be:

```suggestion
		if resp.StatusCode >= 400 && resp.StatusCode <== 499 {
```

Reply:
Correct, but this is now the previous `else` branch as the linter wanted an "early return", so I have to invert the conditional.

Reply:
ah sorry didn't catch the error changed as well when going line by line